3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala

```diff
@@ -231,7 +235,25 @@ object EnrichmentManager {
       event.page_urlquery = components.query.orNull
       event.page_urlfragment = components.fragment.orNull
     }
-
+    
+    // For parsing ValidationString (value) to String type
+    def tryParse(s: scalaz.Validation[String,String]):String = {
+      s match {
+        case Success(pass) => {
+          pass
+        }
+        case Failure(failure) => {
+          failure
+        }
+      }
+    }
```

Comment:
This function ^^ is downsampling again. We're losing the essential information about whether we have a success or failure.